### PR TITLE
Analizo module names

### PR DESCRIPTION
**Pull request by rafamanzo**
Refactored Analizo Collector into different classes as MetricFu was.

The Analizo parser now generates module names with the class name. So we are able to identify files with two classes.

### CODE REVIEW

File: lib/metric_collector/native/analizo/collector.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        def initialize
+          super("Analizo",
+                YAML.load_file("#{Rails.root}/lib/metric_collector/native/descriptions.yml")["analizo"],
+                parse_supported_metrics)
+        end
+
+        def collect_metrics(code_directory, wanted_metric_configurations, processing)
+          self.wanted_metrics = wanted_metric_configurations
+          self.processing = processing
+
```
Comment (diegoamc) [inline lib/metric_collector/native/analizo/collector.rb]: Do we need those assignments above?

File: lib/metric_collector/native/analizo/parser.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            list.delete_at(list.size - 1)
+          end
+          list.join(".")
+        end
+
+        def parse_file_name(file_name)
+          file_name_tokenized = file_name.to_s.split("/")
+          file_name_tokenized[file_name_tokenized.size - 1] = remove_extension(file_name_tokenized.last)
+          file_name_tokenized.join(".")
+        end
```
Comment (diegoamc) [inline lib/metric_collector/native/analizo/parser.rb]: What if there is a dot on the file_name?
See https://github.com/mezuro/kalibro_processor/blob/master/lib/metric_collector/native/metric_fu/parser/flog.rb

File: lib/metric_collector/native/analizo/collector.rb (hunk 1 above)
Comment (rafamanzo) [inline lib/metric_collector/native/analizo/collector.rb]: Look at MetricCollector::Base. The setters are overridden in order to parse those attributes. That's why we assign them and then use.

File: lib/metric_collector/native/analizo/parser.rb (hunk 2 above)
Comment (rafamanzo) [inline lib/metric_collector/native/analizo/parser.rb]: Agreed.

**Pull request merged**